NVFuser fails to build in some environment

https://github.com/NVIDIA/Fuser/pull/4565

Quote from @crcrpar https://github.com/NVIDIA/Fuser/pull/4565#issuecomment-3031201910 :

I'm currently root causing but on my local env with cuda 12.8, source install is failing as follows:
```
FAILED: CMakeFiles/nvf_py_internal.dir/python/python_frontend/python_bindings.cpp.o
/usr/bin/c++ -DC10_BUILD_MAIN_LIB=1 -DEXTENSION_NAME=_C -DNVFUSER_DISTRIBUTED -DTORCH_CUDA_BUILD_MAIN_LIB -DUSE_C10D_GLOO -DUSE_C10D_NCCL -DUSE_DISTRIBUTED -DUSE_RPC -DUSE_TENSORPIPE -I/home/mkozuki/ghq/github.com/NVIDIA/Fuser/python -I/home/mkozuki/ghq/github.com/NVIDIA/Fuser/python/python_common -I/home/mkozuki/ghq/github.com/NVIDIA/Fuser/csrc -I/home/mkozuki/ghq/github.com/NVIDIA/Fuser/lib/dynamic_type/src -isystem /home/mkozuki/ghq/github.com/NVIDIA/Fuser/third_party/flatbuffers/include -isystem /home/mkozuki/ghq/github.com/crcrpar/torch-1/torch/include -isystem /home/mkozuki/ghq/github.com/crcrpar/torch-1/torch/include/torch/csrc/api/include -isystem /usr/local/cuda-12.8/include -isystem /home/mkozuki/.uv_venvs/py311/include -isystem /home/mkozuki/.local/share/uv/python/cpython-3.11.12-linux-x86_64-gnu/include/python3.11 -Wno-psabi  -O3 -DNDEBUG -std=gnu++20 -fPIC -fvisibility=hidden -fvisibility-inlines-hidden -Wall -Wno-unused-function -Werror -MD -MT CMakeFiles/nvf_py_internal.dir/python/python_frontend/python_bindings.cpp.o -MF CMakeFiles/nvf_py_internal.dir/python/python_frontend/python_bindings.cpp.o.d -o CMakeFiles/nvf_py_internal.dir/python/python_frontend/python_bindings.cpp.o -c /home/mkozuki/ghq/github.com/NVIDIA/Fuser/python/python_frontend/python_bindings.cpp
In file included from /home/mkozuki/ghq/github.com/NVIDIA/Fuser/python/python_frontend/python_bindings.cpp:20:
/home/mkozuki/ghq/github.com/NVIDIA/Fuser/csrc/fusion_profiler.h:15:10: fatal error: cupti.h: No such file or directory
   15 | #include <cupti.h>
      |          ^~~~~~~~~
compilation terminated.
[290/333] Building CXX object CMakeFiles/codegen_internal.dir/csrc/serde/fusion_record.cpp.o
[291/333] Building CXX object CMakeFiles/nvf_py_internal.dir/python/python_frontend/multidevice_bindings.cpp.o
ninja: build stopped: subcommand failed.
Configuring CMake with cmake -DCMAKE_PREFIX_PATH=/home/mkozuki/ghq/github.com/crcrpar/torch-1/torch/share/cmake -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=/home/mkozuki/ghq/github.com/NVIDIA/Fuser/python/nvfuser_common -DNVFUSER_CPP_STANDARD=20 -DUSE_DISTRIBUTED=True -DNVFUSER_BUILD_WITH_ASAN=OFF -DNVFUSER_STANDALONE_BUILD_WITH_UCC=OFF -DNVFUSER_EXPLICIT_ERROR_CHECK=OFF -DBUILD_TEST=OFF -DBUILD_PYTHON=ON -DPython_EXECUTABLE=/home/mkozuki/.uv_venvs/py311/bin/python3 -DBUILD_NVFUSER_BENCHMARK=OFF -DNVFUSER_DISTRIBUTED=ON -DUSE_SYSTEM_NVTX=ON -DUSE_HOST_IR_JIT=OFF -B /home/mkozuki/ghq/github.com/NVIDIA/Fuser/python/build -G Ninja ..
```

The pytorch is also built from source in my env fwiw, I'm using uv, and I see cupti.h in the env:
```
% find /usr -name cupti.h
/usr/local/cuda-12.8/extras/CUPTI/include/cupti.h
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -3,6 +3,7 @@
 # SPDX-License-Identifier: BSD-3-Clause
 cmake_minimum_required(VERSION 3.18 FATAL_ERROR)
 project(nvfuser)
+enable_language(CUDA)
 
 cmake_policy(SET CMP0063 NEW) # make symbol visibility always apply
 
@@ -62,6 +63,7 @@ string(APPEND CMAKE_CXX_FLAGS " -Wno-psabi")
 find_package(Torch REQUIRED)
 find_package(Python REQUIRED Development.Module Interpreter)
 find_package(pybind11 REQUIRED)
+find_package(CUDAToolkit REQUIRED)
 
 # need this since the pytorch execution uses a different name
 set(PYTHON_EXECUTABLE ${Python_EXECUTABLE})
@@ -79,11 +81,6 @@ if(TORCH_CUDA_ARCH_LIST)
 endif()
 
 add_subdirectory(${CMAKE_CURRENT_LIST_DIR}/lib/dynamic_type)
-
-# TODO: fix MSVC
-if(NOT MSVC)
-  find_library(LIBCUPTI libcupti.so PATHS ${CUDA_TOOLKIT_ROOT_DIR}/extras/CUPTI/lib64/ ${CUDA_TOOLKIT_ROOT_DIR}/lib64/)
-endif()
 
 # ------------------------------
 # build nvfuser_codegen library
@@ -384,7 +381,6 @@ target_include_directories(codegen_internal PUBLIC ${NVFUSER_PYTHON_COMMON})
 target_include_directories(codegen_internal SYSTEM PUBLIC
   ${CMAKE_SOURCE_DIR}/third_party/flatbuffers/include
   PRIVATE
-  ${CUDA_TOOLKIT_ROOT_DIR}/extras/CUPTI/include
   ${CUDA_INCLUDE_DIRS}
 )
 target_include_directories(codegen_internal PUBLIC
@@ -409,7 +405,7 @@ set_target_properties(codegen_internal PROPERTIES
 list(FILTER TORCH_LIBRARIES EXCLUDE REGEX "libcuda\.so")
 target_link_libraries(codegen_internal PUBLIC
   dynamic_type
-  ${LIBCUPTI}
+  CUDA::cupti
   ${TORCH_LIBRARIES}
   dl
 )
@@ -432,7 +428,7 @@ target_include_directories(nvfuser_codegen PUBLIC
 target_link_libraries(nvfuser_codegen PRIVATE
   flatbuffers
   ${CUDA_NVRTC_LIB}
-  ${LIBCUPTI}
+  CUDA::cupti
   ${TORCH_LIBRARIES}
   dl
 )

